test/test_profiler.py

```diff
@@ -1234,6 +1234,45 @@ def test_utils_compute_self_time(self):
                     for child in event_key.event.children
                 ]))
 
+    def test_utils_compute_queue_depth(self):
+
+        def format_queue_depth(queue_depth_list, events):
+            res = ""
+            for data, event in zip(queue_depth_list, events):
+                res += f"{data.queue_depth} [{event.name()}]\n"
+            return res
+
+        # We have to use Mock because time series data is too flakey to test
+        profiler = self.generate_mock_profile()
+        basic_evaluation = _utils.BasicEvaluation(profiler)
+        self.assertExpectedInline(
+            format_queue_depth(basic_evaluation.queue_depth_list,
+                               basic_evaluation.cuda_events), """\
+1 [cudaLaunchKernel]
+2 [cudaLaunchKernel]
+3 [cudaLaunchKernel]
+2 [GPU]
+1 [GPU]
+0 [GPU]
+""")
+        self.assertExpectedInline(
+            format_queue_depth([
+                basic_evaluation.metrics[k]
+                for k in basic_evaluation.event_keys
+            ], basic_evaluation.events), """\
+0 [CPU]
```

Comment:
I think it would be helpful to either give these more descriptive names or interleave the cuda events. For the first check it's pretty clear what's going on, but for this one it's less clear how to check for correctness.

Reply:
I annotate the cpu event name if that helps.

Reply:
I think it's more that there's no way to check the correctness of the interleaving at a glance. (Because it depends on the specific timings.)

Reply:
Actually, scratch that. The more descriptive names make it super clear exactly what's going on.